lib/galaxy/jobs/runners/cli.py

```diff
@@ -121,6 +121,29 @@ def queue_job(self, job_wrapper):
         # Add to our 'queue' of jobs to monitor
         self.monitor_queue.put(ajs)
 
+    def submit(self, shell, job_interface, job_file, galaxy_id_tag, retry=MAX_SUBMIT_RETRY, timeout=10):
+        """
+        Handles actual job script submission.
+
+        If submission fails will retry `retry` time with a timeout of `timeout` seconds.
+        Retuns the returncode of the submission and the stdout, which contains the external job_id.
+        """
+        cmd_out = shell.execute(job_interface.submit(job_file))
+        if cmd_out.returncode == 0:
+            return cmd_out.returncode, cmd_out.stdout
+        stdout = '(%s) submission failed (stdout): %s' % (galaxy_id_tag, cmd_out.stdout)
+        stderr = '(%s) submission failed (stderr): %s' % (galaxy_id_tag, cmd_out.stderr)
+        log_func = log.warning if retry > 0 else log.error
```

Comment:
I'd make this log.debug, personally. I.e. "please don't bother me with this if it will work eventually, I only care if it fails at the end"

Reply:
also given that you have exactly the same if case below `retry > 0`, it seems odd to have `log_func` with different behaviours that precisely match the below branching. Might as well just replace it with the full `log....()` calls